crawl4ai/utils.py

```
@@ -1991,24 +1991,108 @@ def fast_format_html(html_string):
+    Args:
+        href (str): The URL to normalize
+        base_url (str): The base URL to use for relative URLs
+        
+    Returns:
+        str: The normalized and cleaned URL
+    """
 
-    # Use urljoin to handle all cases
-    normalized = urljoin(base_url, href.strip())
-    return normalized
+def normalize_url(href, base_url):
```

Ruff (0.8.2)

Redefinition of unused `normalize_url` from line 2026

(F811)